bugsnag-plugin-android-ndk/src/main/jni/ctjournal.c

```diff
@@ -0,0 +1,622 @@
+#include "bugsnag_ndk.h"
+#include "utils/string.h"
+#include <errno.h>
+#include <fcntl.h>
+#include <math.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+/* Crash-time journal writing functions. These must be async-safe and fast,
+ * which is why they use low-level open() and write() for everything, and make
+ * no calls to formatted output functions like sprintf.
+ *
+ * Journal writing functions end writing just after the opening quote whenever
+ * strings are involved. So for example, a function that leads up to writing a
+ * map key will end just after the '"', not before. The same goes for higher
+ * level string writing functions (not including write_string, write_raw_string,
+ * write_raw_bytes, and write_raw_literal).
+ *
+ * There are two versions of some functions to handle the requirement to close a
+ * quoted string, or to set up for a map.
+ */
+
+/* IMPORTANT: Keep this consistent with:
+ * - bugsnag_breadcrumb_type in
+ *       bugsnag-plugin-android-ndk/src/main/assets/include/event.h
+ * - BreadcrumbType in
+ *       bugsnag-android-core/src/main/java/com/bugsnag/android/BreadcrumbType.kt
```

Comment:
This doesn't seem ideal - is there any way to enforce keeping these in sync?

Reply:
I can't think of any, short of some kind of multi-language code generator...

`event.h` already had synchronization requirements with `BreadcrumbType.kt` before. There is also other existing code with similar issues (any place we use enums that have JVM analogues).

Reply:
I guess we're stuck with this solution - at least the types of breadcrumbs should change very rarely, if at all.

Reply:
Yeah, the types have never changed.